py/dml/c_backend.py

```diff
@@ -1648,8 +1693,18 @@ def generate_deinit(device):
                 out('}\n', preindent=-1)
 
         with LogFailure(device.site, device, ()):
-            codegen_inline_byname(device, (), 'destroy', [], [],
-                                  device.site).toc()
+            code = codegen_inline_byname(device, (), 'destroy', [], [],
+                                         device.site)
+        if not code.is_empty:
+            # TODO to consider: move the device entry to before _cancel_all
+            # of events are called? This would mean any immediate afters of
+            # events' destroy calls will be delayed until after destroy on the
+            # device is called, instead of being executed once the event's
+            # destroy would return.
```

Comment:
oh wait the `_cancel_all` inline call doesn't generate device entries and exits. I figured it would use the wrapped `__destroy` event methods, but it doesn't. That's a bug.

Reply:
N/A anymore with the `VT_stacked_post` based implementation